doc/reverse_engineering/game_mechanics/damage.md

```diff
@@ -15,63 +16,38 @@ villager        | villager | 3             | 0              | 2               |
 trebuchet       | castle   | 200           | 0              | 323             | 431            | 539
 ram             | castle   | 2             | 0              | 88              | 117            | 146
 
+## Attack damage formula
 
-## Attack damage calculation theory
-
-Each unit has a list of attack and armor classes.
-
-When some unit attacks another, each attacker's attack classes is used only if the target has the matching armor class.
-
-For example: **Halberdier vs Marmeluke**
-
-Halberdier has attack classes and values:
-* class 4 (Melee): 6
-* class 5 (War Elephants): 28
-* class 8 (Cavalry): 32
-* class 16 (Ships, Camels, Saboteurs): 16
-* class 21 (Buildings): 1
-* class 29 (Eagle Warrior): 1
-
-Mameluke has armor classes and values:
-* class 3 (Ranged Attack): 0
-* class 4 (Melee): 1
-* class 8 (Cavalry): 11
-* class 15 (Archers): 0
-* class 16 (Ships, Camels, Saboteurs): 0
-* class 19 (Special Units): 0
-
+The damage in AoE1 and AoE2 is calculated with the following formula
 
-In this example, the classes 4, 8 and 16 are used.
-
-Damage is calculated as follows:
-
-``` python
-
-overall_damage = 0
-for attack_damage, armor_absorbtion in matching_attack_classes(attacker, attacked):
-    damage = attack_damage - armor_absorbtion
-    if damage > 0:
-        overall_damage += damage
-
-if overall_damage < 1:
-    overall_damage = 1
-
-return overall_damage
 ```
+damage = max(1, ((max(0, melee_damage - melee_armor)
+                + max(0, pierce_damage - pierce_armor)
+                + sum(max(0, attack_bonus - attack_resist)))
+                * elevation_factor) * stray_factor)
+```
+
+Explanation:
 
-So in the example we have:
-`(6 - 1) + (32 - 11) + (16 - 0) = 5 + 21 + 16 = 42`
+* `melee_damage`: Melee damage of the attacking unit
+* `melee_armor`: Melee armor of the defending unit
+* `pierce_damage`: Pierce damage of the attacking unit
+* `pierce_armor`: Pierce armor of the defending unit
+* `attack_bonus`: Attack bonus (flat additive) against a target's armor class.
+* `attack_resist`: Attack resist (flat additive) against an attack bonus.
```

Comment:
we have to clarify how those values are gathered, i.e. the attack/armor classes and boni.

Reply:
should be included now.